crunchers/test.cc

```diff
@@ -242,98 +479,28 @@ TEST(CTFOCrunchersTest, AutogeneratedTopCardsStorageIsUpToDate) {
   }
 }
 
-using CTFOTopCardsResponse = CTFO::CruncherResponse<CTFO::TopCardsCruncherResponse>;
+const std::vector<CTFO::TopCardsCruncherArgs>& GetTopCardsCruncherCoplexTestArgs();
+void TopCardsCruncherComplexTestCheck(const std::string& base_url);
 
-CTFOTopCardsResponse GetTopCards(const std::string& url, uint64_t ind) {
-  const auto response = HTTP(GET(url + "/data?i=" + current::ToString(ind)));
-  EXPECT_EQ(HTTPResponseCode.OK, response.code);
-  return ParseJSON<CTFOTopCardsResponse>(response.body);
-}
-
-TEST(CTFOCrunchersTest, TopCardsCruncherLocalTest) {
+TEST(CTFOCrunchersTest, TopCardsCruncherLocalTestComplex) {
   const std::string golden_db_file_name = current::FileSystem::JoinPath("golden", "top_cards_db.json");
   CTFO_Local::Sherlock local_stream(CTFO::SchemaKey(), golden_db_file_name);
 
-  using CTFOTopCardsCruncher = CTFO::TopCardsCruncher<CTFO_Local>;
-
-  std::vector<CTFO::TopCardsCruncherArgs> args;
-  const auto calculator = [](const CTFO::CardCounters& counters) -> double {
-    return 160000 * counters.ctfo + 8000 * counters.tfu + 400 * (counters.fav - counters.unfav) +
-           20 * counters.skip + counters.seen;
-  };
-  for (uint32_t i = 0; i < 12; ++i) {
-    args.emplace_back(std::chrono::seconds(i + 1), 20, calculator);
-  }
-  for (uint32_t i = 0; i < 15; ++i) {
-    args.emplace_back(std::chrono::minutes(i + 1), 20, calculator);
-  }
-  args.emplace_back(std::chrono::hours(3600 * 21), 20, calculator);
-
   const std::string base_url = Printf("http://localhost:%u/top_cards", FLAGS_cruncher_http_test_port);
-  auto top_cards_cruncher =
-      std::make_unique<CTFOTopCardsCruncher>(FLAGS_cruncher_http_test_port, "/top_cards", args);
+  CTFO::TopCardsCruncher<CTFO_Local> top_cards_cruncher(
```

Comment:
It doesn't want to compile, because `HTTPRoutesScope` (it is a member of the `GenericCruncherImpl`) has a deleted copy constructor.